src/Nethermind/Nethermind.Consensus.Clique/CliqueBlockProducer.cs

```diff
@@ -388,7 +388,7 @@ public async Task StopAsync()
             var selectedTxs = _txSource.GetTransactions(parentBlock.Header, header.GasLimit);
             Block block = new Block(header, selectedTxs, new BlockHeader[0]);
             header.TxRoot = new TxTrie(block.Transactions).RootHash;
-            block.Header.Author = _address;
+            block.Header.Author = _signer.SigningAddress;
```

Comment:
same here -> sealer/validator is the author
and it happens to be that it is the sealer that should be signing

Reply:
we split the responsibility on the implementation level but confused it on the semantical level